### ISSUE TITLE
dbg_macro not caught when in submodule

### ISSUE THREAD

**Issue by vmalloc** [rust-lang/rust-clippy#7275]
Lint name: dbg_macro

I tried this code:

```rust
// src/main.rs
#![deny(warnings)]
#![deny(clippy::dbg_macro)]

mod sub;

fn main() {
    sub::func();
}


// src/sub.rs
pub fn func() {
    dbg!(10);
}

```

I expected the dbg! macro to be caught when running `cargo clippy`.

Instead, the lint does not trigger, and clippy finishes successfully.

The surprising bit (perhaps connected to #4612 ?) is when running clippy after forcefully touching the file (e.g. `find . -name "*.rs" -exec touch {}`), clippy catches it just fine 🤷 

### Meta

- `cargo clippy -V`: clippy 0.1.52 (9bc8c42b 2021-05-09)
- `rustc -Vv`:
  ```
  rustc 1.52.1 (9bc8c42bb 2021-05-09)
binary: rustc
commit-hash: 9bc8c42bb2f19e745a63f3445f1ac248fb015e53
commit-date: 2021-05-09
host: x86_64-apple-darwin
release: 1.52.1
LLVM version: 12.0.0
  ```

**Comment by ralpha** [comment]
I had the same problem with `dbg_macro`. (And maybe others.)
Even when using `#![deny(clippy::all)]` it did not report an error.
(rust: 1.56.1, Linux)
